src/vm/jithelpers.cpp

```diff
@@ -5591,15 +5591,52 @@ HCIMPL0(VOID, JIT_PollGC)
 }
 HCIMPLEND
 
+
+/*************************************************************/
+// This helper is similar to JIT_RareDisableHelper, but has more operations
+// tailored to the post-pinvoke operations.
+extern "C" FCDECL0(VOID, JIT_PInvokeEndRarePath);
+
+HCIMPL0(void, JIT_PInvokeEndRarePath)
+{
+    BEGIN_PRESERVE_LAST_ERROR;
+
+    FCALL_CONTRACT;
+
+    Thread *thread = GetThread();
+
+    // We need to disable the implicit FORBID GC region that exists inside an FCALL
+    // in order to call RareDisablePreemptiveGC().
+    FC_CAN_TRIGGER_GC();
+    thread->RareDisablePreemptiveGC();
+    FC_CAN_TRIGGER_GC_END();
+
+    FC_GC_POLL_NOT_NEEDED();
+
+    HELPER_METHOD_FRAME_BEGIN_NOPOLL();    // Set up a frame
+    thread->HandleThreadAbort();
+    HELPER_METHOD_FRAME_END();
+
+    InlinedCallFrame* frame = (InlinedCallFrame*)thread->m_pFrame;
+
+    thread->m_pFrame->Pop(thread);
+
+    frame->m_pCallerReturnAddress = NULL;
```

Comment:
Do you actually need to clear the caller address after unlinking the frame? Nothing should care about the content of unlinked frame. You can set it to garbage value in debug builds only to be sure.

Reply:
I don't know. I saw it was done by the JIT in lower.cpp.

Reply:
JIT does not unlink the frame right away. It delays it till the end of the method. It is why it needs to clear the caller.

Reply:
So sounds like we can clear it in this helper right?

Reply:
I’ll remove it and update the PR

Reply:
It should be removed from the assembly stubs too - it is where it matters the most because of they are the hot path.

Reply:
Ah, good point. I forgot about the stubs.